### PR TITLE
gcr kritis signer

### LINKED ISSUE #586: Automated check and sign on completion of an analysis of an image in GCR
**Is your feature request related to a problem? Please describe.**
I want to be able to automatically add an attestation for any GCR image after its vulnerability scan is completed.

**Describe the solution you'd like**

- a cloud run service
- subscribing to the topic `container-analysis-occurrences-v1`
- on each DISCOVERY occurrence in the status completed, validate the scan against the policy

**Describe alternatives you've considered**
I considered writing a little cloud run service to handle the pubsub event and in turn, invoking the signer. This would result in a bigger container runtime.

**Additional context**

### PR DESCRIPTION
**Pull request by mvanholsteijn**
A standalone server to create attestations for images on request, or automatically  after the images are scanned. 

```
gcr-signer 
  -attestation_project string
        project id for GCP project that stores attestation, use image project if set to empty
  -kms_digest_alg string
        kms digest algorithm, must be one of SHA256|SHA384|SHA512, and the same as specified by the key version's algorithm
  -kms_key_name string
        kms key name, in the format of in the format projects/*/locations/*/keyRings/*/cryptoKeys/*/cryptoKeyVersions/*
  -note_name string
        note name that created attestations are attached to, in the form of projects/[PROVIDER_ID]/notes/[NOTE_ID]
  -overwrite
        overwrite attestation if already existed
  -policy string
        vulnerability signing policy file path, e.g., /tmp/vulnz_signing_policy.yaml
  -vulnz_timeout string
        timeout for polling image vulnerability , e.g., 600s, 5m (default "5m")
```

Read docs/gcr-signer.md for the full details.

close #586

### CODE REVIEW

**Review by ooq — COMMENTED** [review]
Hi @mvanholsteijn, I did a run on the tutorial, works great! I think this PR is good to go once you address the comments (all on the tutorial).

File: docs/gcr-signer.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+        ```shell
+             gcloud services enable run.googleapis.com
+
+             gcloud beta run deploy gcr-kritis-signer \
+              --image  gcr.io/${PROJECT_ID}/gcr-kritis-signer:latest \
+              --service-account ${SA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com \
+              --set-env-vars "ATTESTATION_POLICY=$(cat samples/signer/policy.yaml)" \
+              --timeout 2m \
+              --allow-unauthenticated \
```
[the remark this answers is not in the document]
Comment (mvanholsteijn) [reply docs/gcr-signer.md:235]: added a remark. will do later.

**Review by ooq — APPROVED** [review]
Thanks for the PR! LGTM.

**Comment by mvanholsteijn** [comment]
@ooq thank you for the merge!

Will the gcr-kritis-signer container image be available from the kritis project? If so, when?

**Comment by ooq** [comment]
@mvanholsteijn Do you mean to release the image in a gcr.io repository or other alternatives? We currently don't release images simply because we don't have a documented process for that.

**Comment by mvanholsteijn** [comment]
that is indeed what I ment.

**Pull request merged**